---
title: visiumStitched_brain
output:
  github_document:
    html_preview: true
  html_document:
    toc: true
    toc_flot: true
    includes:
      in_header: header.html
      after_body: footer.html
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r 'setup', include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "img/",
    out.width = "100%"
)
```

## Overview

This repository contains the data used to demonstrate the utility of [`visiumStitched`](https://github.com/LieberInstitute/visiumStitched). Three [Visium](https://www.10xgenomics.com/products/spatial-gene-expression)
capture areas were collected from the postmortem human brain from one donor. We use
[Fiji](https://imagej.net/software/fiji/) and `visiumStitched` to stitch the gene-expression and imaging
data into one sample for downstream analysis.

If you tweet about this website, the data or the R package please use
the <code>\#visiumStitched</code> hashtag. You can find previous tweets
that way as shown
<a href="https://twitter.com/search?q=%23visiumStitched&src=typed_query">here</a>.

Thank you for your interest in our work!

## Study Design

TODO: add figure 1 and caption

## Internal

* JHPCE location: `/dcs05/lieber/lcolladotor/visiumStitched_LIBD1070/visiumStitched_brain`.

The code in this repository follows the organization from https://github.com/LieberInstitute/template_project following the guidelines from [organizing-your-work](https://lcolladotor.github.io/bioc_team_ds/organizing-your-work.html#.YzL43uzMKX0).

## Introductory Material

Check out [this guide](http://research.libd.org/visiumStitched/articles/full_demo.html) to using
`visiumStitched`, which enabled the integration of all three capture areas for this project.

<iframe width="560" height="315" src="https://www.youtube.com/embed/kFLtpK3qbSY?si=u6bIKY5U9yccx62R" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen data-external="1"></iframe>

## Interactive Websites

Explore the [Shiny app](https://libd.shinyapps.io/visiumStitched_brain), powered by
[spatialLIBD](https://doi.org/10.1186/s12864-022-08601-w).

Alternatively, you may choose to run the app locally with [`spatialLIBD::run_app()`](http://research.libd.org/spatialLIBD/reference/run_app.html),
which gives more freedom regarding memory usage and session length. See [this example](https://github.com/LieberInstitute/visiumStitched_brain/blob/devel/code/05_shiny/app.R)
for invoking `spatialLIBD::run_app()` with an appropriate set of arguments.

## R Objects

The transcriptomics data, SpaceRanger outputs, and Fiji outputs are all available
through `spatialLIBD::fetch_data()`. A comprehensive vignette using these objects
is written [here](http://research.libd.org/visiumStitched/articles/full_demo.html).

```{r "pull_data", fig.height = 8, fig.width = 9}
## Check that you have a recent version of spatialLIBD installed
stopifnot(packageVersion("spatialLIBD") >= "1.17.6")

## Download the spot-level data, which is a SpatialExperiment object
spe <- spatialLIBD::fetch_data(type = "visiumStitched_brain_spe")
spe

## Show clustering results from PRECAST at k = 2
vis_clus(spe, clustervar = 'precast_k2', is_stitched = TRUE)
```

## Citing Our Work

TODO: manuscript citation

### Cite `spatialLIBD`

Below is the citation output from using `citation('spatialLIBD')` in R. Please run this yourself
to check for any updates on how to cite __spatialLIBD__.

```{r 'citation_spatialLIBD'}
print(citation("spatialLIBD")[1], bibtex = TRUE)
```

Please note that `spatialLIBD` was only made possible thanks to many other R and bioinformatics
software authors, which are cited either in the vignettes and/or the paper(s) describing the
package.
